inst/include/model_simulation.hpp

```diff
@@ -131,5 +146,213 @@ void run_new_infections(int hiv_step,
+        for (int hu = 0; hu < pars.treatment_stages; hu++) {
+          intermediate.deaths_art =
+              pars.art_mortality(hu, hm, ha, g) * pars.artmx_timerr(hu, time_step) *
+              state_next.art_strat_adult(hu, hm, ha, g);
+          intermediate.hiv_deaths_age_sex(ha, g) += pars.dt * intermediate.deaths_art;
+          state_next.aids_deaths_art(hu, hm, ha, g) += pars.dt * intermediate.deaths_art;
+          intermediate.gradART(hu, hm, ha, g) = -intermediate.deaths_art;
+        }
+
+        for (int hu = 0; hu < (pars.treatment_stages - 1); hu++) {
+          intermediate.gradART(hu, hm, ha, g) +=
+              -state_next.art_strat_adult(hu, hm, ha, g) / pars.h_art_stage_dur(hu);
+          intermediate.gradART(hu + 1, hm, ha, g) +=
+              state_next.art_strat_adult(hu, hm, ha, g) / pars.h_art_stage_dur(hu);
+        }
+
+        // ART dropout
+        if (pars.art_dropout(time_step) > 0) {
+          for (int hu = 0; hu < pars.treatment_stages; hu++) {
+            intermediate.grad(hm, ha, g) += pars.art_dropout(time_step) * state_next.art_strat_adult(hu, hm, ha, g);
+            intermediate.gradART(hu, hm, ha, g) -=
+                pars.art_dropout(time_step) * state_next.art_strat_adult(hu, hm, ha, g);
+          }
+        }
+      }
+    }
+  }
+}
+
+template<typename real_type>
+void run_art_initiation(int hiv_step,
+                        int time_step,
+                        const Parameters<real_type> &pars,
+                        const State<real_type> &state_curr,
+                        State<real_type> &state_next,
+                        IntermediateData<real_type> &intermediate) {
+  for (int g = 0; g < pars.num_genders; ++g) {
+    intermediate.Xart_15plus = 0.0;
+    intermediate.Xartelig_15plus = 0.0;
+    intermediate.expect_mort_artelig15plus = 0.0;
+    for (int ha = pars.hIDX_15PLUS; ha < pars.age_groups_hiv; ++ha) {
+      for (int hm = intermediate.everARTelig_idx; hm < pars.disease_stages; ++hm) {
+        if (hm >= intermediate.anyelig_idx) {
+          // TODO: Implement special population ART eligibility
+          real_type prop_elig = 1.0;
+          intermediate.artelig_hahm(hm, ha - pars.hIDX_15PLUS) = prop_elig * state_next.hiv_strat_adult(hm, ha, g);
+          intermediate.Xartelig_15plus += intermediate.artelig_hahm(hm, ha - pars.hIDX_15PLUS);
+          intermediate.expect_mort_artelig15plus +=
+              pars.cd4_mortality(hm, ha, g) * intermediate.artelig_hahm(hm, ha - pars.hIDX_15PLUS);
+        }
+        for (int hu = 0; hu < pars.treatment_stages; ++hu)
+          intermediate.Xart_15plus +=
+              state_next.art_strat_adult(hu, hm, ha, g) + pars.dt * intermediate.gradART(hu, hm, ha, g);
+      }
+    }
+
+    // calculate number on ART at end of ts, based on number or percent
+    if (pars.dt * (hiv_step + 1) < 0.5) {
+      if ((!pars.art15plus_isperc(g, time_step - 2)) & (!pars.art15plus_isperc(g, time_step - 1))) {
```

Comment:
```suggestion
      if ((!pars.art15plus_isperc(g, time_step - 2)) && (!pars.art15plus_isperc(g, time_step - 1))) {
```

and elsewhere where you use `&` or `|` in logical contexts (e.g., line 222 below)